src/System.Runtime.Serialization.Xml/tests/Performance/PerformanceTestsCommon.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Xunit.Performance;
+using System.Xml;
+using SerializationTypes;
+
+namespace System.Runtime.Serialization
+{
+    #region Performance tests configuration types
+
+    public enum SerializerType
+    {
+        DataContractSerializer,
+        DataContractJsonSerializer,
+        XmlSerializer,
+        JsonNet
+    }
+
+    public enum TestType
+    {
+        SimpleType,
+        ISerializable,
+        Dictionary,
+        ListOfInt,
+        String,
+        ByteArray,
+        XmlElement
+    }
+
+    public interface IPerfTestSerializer
+    {
+        void Init(object obj);
+        void Serialize(object obj, Stream s);
+        void Deserialize(Stream s);
+    }
+
+    public abstract class SerializerFactory
+    {
+        public abstract IPerfTestSerializer GetSerializer(SerializerType serializerType);
+    }
+
+    #endregion
+
+    public class PerformanceTestCommon
+    {
+
+        #region Performance test configuration
+
+        public static IEnumerable<object[]> PerformanceMemberData(SerializerType serializerType)
+        {
+            yield return new object[] { 100, TestType.ByteArray, 1024, serializerType };
+            yield return new object[] { 5, TestType.ByteArray, 1024 * 1024, serializerType };
```

Comment:
1. Each line is a configuration for a scenario.
2. This method would take the serializerType as parameter and we have 4 different serializer types so it needs to be there.
3. Agree, that would be nice. But for me, there was an issue when I did that. Object type when passed into MemberData causes some strange issue that the tests don't get run.